quipucords/tests/scanner/network/test_utils.py

```diff
@@ -1,11 +1,102 @@
 """Test the network scanner utility functions."""
 
 import unittest
+from pathlib import Path
 from unittest import mock
 
+import pytest
+from django.forms import model_to_dict
+
+from api import vault
+from api.models import Credential
+from api.serializers import SourceSerializer
+from constants import GENERATED_SSH_KEYFILE, DataSources
 from scanner.network import utils
+from tests.factories import CredentialFactory, SourceFactory
+
+
+@pytest.mark.django_db
+def test_scan_inventory_with_valid_ssh_key(
+    network_credential, network_host_addresses, faker
+):
+    """Test construct_inventory returns dict with valid path to the SSH key."""
+    with network_credential.generate_ssh_keyfile() as ssh_keypath:
+        cred_data = model_to_dict(network_credential)
+        cred_data[GENERATED_SSH_KEYFILE] = ssh_keypath
+        # network_host_addresses[0] because it has only one address by default.
+        host_address = network_host_addresses[0]
+        _, inventory_dict = utils.construct_inventory(
+            [(host_address, cred_data)],
+            connection_port=faker.pyint(),
+            concurrency_count=faker.pyint(),
+        )
+
+        ssh_keyfile = inventory_dict["all"]["children"]["group_0"]["hosts"][
+            host_address
+        ]["ansible_ssh_private_key_file"]
+        assert Path(ssh_keyfile).exists()
+    # check context manager cleanup
+    assert not Path(ssh_keyfile).exists()
+
+
+@pytest.mark.django_db
+def test_construct_inventory_from_source_and_credential_models(faker):
+    """Test construct_inventory using data from full Source and Credential objects."""
+    # TODO Simplify this test. Does this really need to use full model objects?
```

Comment:
I think we could get rid of the database requirement here by using the factories to create the objects but not saving it to db (`SomeFactory.build(*args, **kwargs)`)

It won't solve the design issue here (mainly because we don't even need the models, just a dict representations of it)

Reply:
Simply swapping in `.build` calls does not completely solve the problem here. So, I will look away and let refactoring this be a problem for _future us_ because this is _mostly_ just the result of transplanting (and slightly updating) an old test. 🙈